spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelAutoConfiguration.java

```diff
@@ -99,13 +99,12 @@ SdkTracerProvider otelTracerProvider(SpanLimits spanLimits, ObjectProvider<List<
 
 	@Bean
 	@ConditionalOnMissingBean
-	Resource otelResource(Environment env, ObjectProvider<List<ResourceProvider>> resourceProviders) {
+	Resource otelResource(Environment env, ObjectProvider<List<Supplier<Resource>>> resourceProviders) {
```

Comment:
I wonder if this won't be a problem with Spring Cloud Stream cc @olegz ?

Reply:
Technically, this should not be a problem, even if those `Suppliers` are beans in the context of s-c-stream. Basically unless explicit bindings are defined against them they are just beans. If they are not beans that s-c-function/stream wouldn't even have a way to know about them so that much better.

But the short answer is "NOT a problem"